flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveSourceSplit.java

```diff
@@ -67,12 +69,22 @@ public HiveSourceSplit(
         super(id, filePath, offset, length, hostnames, readerPosition);
         this.hiveTablePartition =
                 checkNotNull(hiveTablePartition, "hiveTablePartition can not be null");
+        final String hosts =
+                hostnames.length == 0 ? "(no host info)" : " hosts=" + Arrays.toString(hostnames);
+        this.detailInfoForSplit =
+                String.format(
+                        "HiveSourceSplit: %s [%d, %d) %s ID=%s position=%s",
+                        filePath, offset, offset + length, hosts, id, readerPosition);
     }
 
     public HiveTablePartition getHiveTablePartition() {
         return hiveTablePartition;
     }
 
+    public String getDetailInfoForSplit() {
+        return detailInfoForSplit;
```

Comment:
We can just use `toString` to print the split. That'll make things more consistent. And I think information such as `hosts` and `offset` are not very useful because they're irrelevant to how the split is consumed.